kornia/filters/bilateral.py

```diff
@@ -0,0 +1,63 @@
+import math
+
+import numpy as np
+
+
+# gaussian filter
+def gfunc(x,y,sigma):
+    return (math.exp(-(x**2 + y**2)/(2*(sigma**2))))/(2*3.14*(sigma**2))
+
+def gaussFilter(size, sigma):
+    out = np.zeros(size)
+    for i in range(size[0]):
+        for j in range(size[1]):
+            out[i,j] = gfunc(i-size[0]//2,j-size[1]//2, sigma )
+    return out/np.sum(out)
+
+def bfunc(i,j,fw,fh,image,sigma1, bilateralWFilter):
+    imgwork = image[i - fh//2:i+1 + fh//2, j - fw//2:j+1 + fw//2, :]
+
+    bilateralIFilter = ((imgwork - image[i, j,:])**2 )/(2*(sigma1**2))
+
+    bilateralFilter = np.exp(-1*bilateralIFilter)*bilateralWFilter
+    bilateralFilter = bilateralFilter/np.sum(bilateralFilter,axis=(0,1))
+    return np.sum(np.multiply(imgwork, bilateralFilter),axis=(0,1))
+
+def bilateral_blur(image,  kernel_size, sigma):
+        size = image.shape
+        fw, fh = kernel_size[0], kernel_size[1]
+        sigma1, sigma2 = sigma[0], sigma[1]
+        bilateral1 = 2*3.14*sigma2*sigma2*gaussFilter((fw,fh), sigma2)
+        if len(image.shape) < 3  or image.shape[2] == 1:
+            bilateralWFilter = np.resize(bilateral1,(*bilateral1.shape,1))
+        else:
+            bilateralWFilter = np.stack([bilateral1, bilateral1, bilateral1], axis=2)
+        out = np.zeros((size[0]-2*fw +1,size[1]-2*fh +1,size[2]))
+        for i in range(size[0]-2*fh +1):
```

Comment:
we need to find a proper way to parallelize this for loops /cc @AnimeshMaheshwari22 @lferraz @ducha-aiki